rust/zypp-agama/src/lib.rs

```diff
@@ -26,6 +26,21 @@ pub struct Repository {
     pub user_name: String,
 }
 
+impl Repository {
+    /// check if url points to local repository.
+    /// Can be None if url is invalid
+    pub fn is_local(&self) -> Result<bool, url::ParseError> {
```

Comment:
What about reusing the libzypp functionality from the Url class? https://github.com/openSUSE/libzypp/blob/master/zypp-logic/zypp-core/Url.h#L274